fs/diff.go

```diff
@@ -102,9 +103,6 @@ func Changes(ctx context.Context, a, b string, changeFn ChangeFunc) error {
 	if a == "" {
 		logrus.Debugf("Using single walk diff for %s", b)
 		return addDirChanges(ctx, changeFn, b)
-	} else if diffOptions := detectDirDiff(b, a); diffOptions != nil {
-		logrus.Debugf("Using single walk diff for %s from %s", diffOptions.diffDir, a)
-		return diffDirChanges(ctx, changeFn, a, diffOptions)
 	}
```

Comment:
Why did we remove this? It seems we do not call diffDirChanges from fs.Changes anymore. The same delay seems to be there in nerdctl commit.

Reply:
Because you can't detect dir's underlying filesytem based on string path. 
Before comparing two rootfs, we have information about rootfs's mount.
We can check if the filesystem is overlayfs. If so, we can use new function. Otherwise, we should fallback to `fs.Changes`.

Reply:
Thanks